### PR TITLE
Update some icons to address aliasing on high-res screens

### PR DESCRIPTION
**Pull request by richtabor**
Part of #https://github.com/WordPress/gutenberg/issues/49311, focusing on common blocks/UI icons first. Updates a few icons to remove the aliasing that occurs on high-resolution screens. 

## Why?
We want our icons to look crisp and present well.

## How?
Updating icons within the icons package. This does not change the spirit of the icons, just adjusts the icon paths a little to not use `0.25` X and Y values. 

## Testing Instructions
Check the icons in the following locations: 

1. Heading block. 
2. Gallery block.
3. Grid block.
4. Paragraph block.
5. Row block.
6. Stacked block.
7. List view icon.

## Screenshots or screencast 
CleanShot 2023-03-27 at 16 30 43
CleanShot 2023-03-27 at 16 31 11

### CODE REVIEW

**Comment by jasmussen** [comment]
Really stellar work. Are you also updating and publishing the Figma library as part of this? The gallery code looks slightly more expansive. Can you maybe run that through https://svgomg.net/?

**Comment by richtabor** [comment]
> The gallery code looks slightly more expansive. Can you maybe run that through https://svgomg.net/?

I did with one of the Figma plugins, ran it again. I can't seem to get it without `fill-rule` and `clip-rule` though. The icon comes out filled without those. And ideas how to reduce them?  

> Are you also updating and publishing the Figma library as part of this?

Yes! I have them ready, was waiting for approval first. I'll move these standouts into the actual icon components when we're merged here. 

CleanShot 2023-03-28 at 14 33 13

File: packages/icons/src/library/gallery.js [hunk 1]
```diff
@@ -5,7 +5,11 @@ import { Path, SVG } from '@wordpress/primitives';
 
 export const gallery = (
 	<SVG viewBox="0 0 24 24" xmlns="http://www.w3.org/2000/svg">
-		<Path d="M20.2 8v11c0 .7-.6 1.2-1.2 1.2H6v1.5h13c1.5 0 2.7-1.2 2.7-2.8V8h-1.5zM18 16.4V4.6c0-.9-.7-1.6-1.6-1.6H4.6C3.7 3 3 3.7 3 4.6v11.8c0 .9.7 1.6 1.6 1.6h11.8c.9 0 1.6-.7 1.6-1.6zM4.5 4.6c0-.1.1-.1.1-.1h11.8c.1 0 .1.1.1.1V12l-2.3-1.7c-.3-.2-.6-.2-.9 0l-2.9 2.1L8 11.3c-.2-.1-.5-.1-.7 0l-2.9 1.5V4.6zm0 11.8v-1.8l3.2-1.7 2.4 1.2c.2.1.5.1.8-.1l2.8-2 2.8 2v2.5c0 .1-.1.1-.1.1H4.6c0-.1-.1-.2-.1-.2z" />
+		<Path
+			d="M16.375 4.5H4.625a.125.125 0 0 0-.125.125v8.254l2.859-1.54a.75.75 0 0 1 .68-.016l2.384 1.142 2.89-2.074a.75.75 0 0 1 .874 0l2.313 1.66V4.625a.125.125 0 0 0-.125-.125Zm.125 9.398-2.75-1.975-2.813 2.02a.75.75 0 0 1-.76.067l-2.444-1.17L4.5 14.583v1.792c0 .069.056.125.125.125h11.75a.125.125 0 0 0 .125-.125v-2.477ZM4.625 3C3.728 3 3 3.728 3 4.625v11.75C3 17.273 3.728 18 4.625 18h11.75c.898 0 1.625-.727 1.625-1.625V4.625C18 3.728 17.273 3 16.375 3H4.625ZM20 8v11c0 .69-.31 1-.999 1H6v1.5h13.001c1.52 0 2.499-.982 2.499-2.5V8H20Z"
+			fill-rule="evenodd"
```
Comment (jasmussen) [inline packages/icons/src/library/gallery.js]: It looks like these rules are needed for the gallery icon, I just tested. But they should be `fillRule` and `clipRule` to be react native compatible.

**Review by jasmussen — APPROVED** [review]
This is an awesome attention to detail, thank you. Let's get this in.

Edit: just one small fix first.

File: packages/icons/src/library/gallery.js (hunk 1 above)
Comment (richtabor) [reply packages/icons/src/library/gallery.js]: Good catch! Thanks!

**Pull request merged**